lib/SILGen/SILGen.cpp

```diff
@@ -569,9 +569,15 @@ SILFunction *SILGenModule::emitTopLevelFunction(SILLocation Loc) {
                                    SubstitutionMap(), SubstitutionMap(),
                                    C);
 
+  StringRef name;
+  if (auto overriddenName = getASTContext().LangOpts.entryPointFunctionName) {
+    name = overriddenName.getValue();
+  } else {
+    name = SWIFT_ENTRY_POINT_FUNCTION;
```

Comment:
There are a number of additional usages of `SWIFT_ENTRY_POINT_FUNCTION` that need to be replaced with something like this (e.g. a call to a function that does this check).